docs/en/integrations/language-clients/java/index.md

```diff
@@ -378,6 +378,58 @@ try (PreparedStatement stmt = conn.prepareStatement(
     Assert.assertFalse(rs.next());
 }
 ```
+<br/>
+
+#### Resolving JDBC Timeout on Large Inserts
+
+When performing large inserts in ClickHouse with long execution times, you may encounter JDBC timeout errors like:
+
+```plaintext
+Caused by: java.sql.SQLException: Read timed out, server myHostname [uri=https://hostname.aws.clickhouse.cloud:8443]
+```
+
+These errors can disrupt the data insertion process and affect system stability. To address this issue you need to adjust a few timeout settings in the client's OS.
+
+##### Mac OS
+
+On Mac OS, the following settings can be adjusted to resolve the issue:
+
+- `net.inet.tcp.keepidle`: 60000
+- `net.inet.tcp.keepintvl`: 45000
+- `net.inet.tcp.keepinit`: 45000
+- `net.inet.tcp.keepcnt`: 8
+- `net.inet.tcp.always_keepalive`: 1
+
+##### Linux
+
+On Linux, the equivalent settings alone may not resolve the issue. Additional steps are required due to the differences in how Linux handles socket keep-alive settings. Follow these steps:
+
+1. Adjust the following Linux kernel parameters in `/etc/sysctl.conf` or a related configuration file:
+
+   - `net.inet.tcp.keepidle`: 60000
+   - `net.inet.tcp.keepintvl`: 45000
+   - `net.inet.tcp.keepinit`: 45000
+   - `net.inet.tcp.keepcnt`: 8
+   - `net.inet.tcp.always_keepalive`: 1
+   - `net.ipv4.tcp_keepalive_intvl`: 75
+   - `net.ipv4.tcp_keepalive_probes`: 9
+   - `net.ipv4.tcp_keepalive_time`: 60 (You may consider lowering this value from the default 300 seconds)
+
+2. After modifying the kernel parameters, apply the changes by running the following command:
+
+   ```shell
+   sudo sysctl -p
+   ```
+
+After Setting those settings, you need to ensure that your client enables the Keep Alive option on the socket. Currently, you must use Apache HTTP Client library, as the other two HTTP client libraries in clickhouse-java do not allow setting socket options:
```

Comment:
Optional: Can we move to a separate section with a short guide on changing the default HTTP library?

Reply:
Added. Could you please review it as well?